src/base/bittorrent/torrentimpl.cpp

```diff
@@ -1005,7 +1005,33 @@ QString TorrentImpl::error() const
         return QString::fromStdString(m_nativeStatus.errc.message());
 
     if (m_nativeStatus.flags & lt::torrent_flags::upload_mode)
-        return tr("There's not enough space on disk. Torrent is currently in \"upload only\" mode.");
+    {
+        const auto formatError = [](const QString &error) -> QString
+        {
+            const QString writeErrorStr = tr("Couldn't write to file.");
+            const QString uploadModeStr = tr("Torrent is currently in \"upload only\" mode.");
+
+            return writeErrorStr + QLatin1Char(' ')
+                    + (error.isEmpty() ? QString() : (error + QLatin1Char(' ')))
+                    + uploadModeStr;
+        };
+
+        switch (m_lastFileError.error.value())
+        {
+        case boost::system::errc::read_only_file_system:
+            return formatError(tr("Filesystem is \"read only\"."));
```

Comment:
Why not use `m_lastFileError.error.message()`?
At least on Windows it seems to use platform translated messages: https://stackoverflow.com/questions/17465526/how-to-get-boostsystemerror-codemessage-in-english

Reply:
Done.